rules/xcodeproj.bzl

```diff
@@ -209,8 +209,9 @@ def _xcodeproj_aspect_impl(target, ctx):
 
         swift_module_paths = []
         if SwiftInfo in target:
-            swift_defines.append(depset(target[SwiftInfo].direct_defines))
-            swift_defines.append(target[SwiftInfo].transitive_defines)
+            swift_info = target[SwiftInfo]
+            swift_defines.append(depset(_collect_swift_defines(swift_info.direct_modules)))
+            swift_defines.append(depset(_collect_swift_defines(swift_info.transitive_modules.to_list())))
```

Comment:
`direct_defines` and `transitive_defines` were removed from `rules_swift`. I lifted this approach for collecting defines from Tulsi.

Reply:
So this gives xcode the same invocation as the swift invocation in Bazel? I wonder how it works with `build --features=swift.split_derived_files_generation`, if that is picked up in `transitive_modules` or in any way.

Reply:
I think this is more or less functionally identical to how it was before. `rules_swift` just removed the attributes so now you can only collect them yourself (ex: https://github.com/bazelbuild/tulsi/commit/dc84e3943bf2b11da37e30127e3fa830e293d296)

Reply:
lgtm.

Reply:
For sure, that's probably orthogonal to this! I was just curious if this code caused it to pick it up 2x in any way, it looks like the Xcode argument's aren't matching Bazels.